CRM/Contact/BAO/Query.php

```diff
@@ -516,9 +516,10 @@ public function __construct(
       $this->_fields = array_merge($this->_fields, $fields);
 
       // add activity fields
-      $this->_fields = array_merge($this->_fields, CRM_Activity_BAO_Activity::exportableFields());
-      // Add hack as no unique name is defined for the field but the search form is in denial.
-      $this->_fields['activity_priority_id'] = $this->_fields['priority_id'];
+      $this->_fields = array_merge(
+        $this->_fields,
+        CRM_Activity_BAO_Activity::exportableFields(CRM_Contact_Form_Search::getModeValue($this->_mode)['entity'])
```

Comment:
I think given we only support 'Activity' and 'Case' we probably should stick with hard-coding. I like more metadata but it also feels like it's opening up flakey code to have implied support for more than just the 2 things it does support & there is no particular benefit it NOT defining case activity fields when it isn's a case activity search